assets/wizards/popups/views/segmentation/SingleSegment.js

```diff
@@ -130,16 +133,49 @@ const SegmentsList = ( { segmentId, wizardApiFetch } ) => {
 					description={ __( '(if using WooCommerce checkout)', 'newspack' ) }
 				>
 					<CheckboxControl
-						checked={ is_donor }
-						onChange={ setIsDonor }
+						checked={ segmentConfig.is_donor }
+						onChange={ updateSegmentConfig( 'is_donor' ) }
 						label={ __( 'Has donated', 'newspack' ) }
 					/>
 					<CheckboxControl
-						checked={ is_not_donor }
-						onChange={ setIsNotDonor }
+						checked={ segmentConfig.is_not_donor }
+						onChange={ updateSegmentConfig( 'is_not_donor' ) }
 						label={ __( "Hasn't donated", 'newspack' ) }
 					/>
 				</SegmentSettingSection>
+				<SegmentSettingSection
+					title={ __( 'Category Affinity', 'newspack' ) }
+					description={ __( 'Most read categories of reader.', 'newspack' ) }
+				>
+					{ take(
+						siteCategories.map( category => {
+							const selected = segmentConfig.favorite_categories;
+							const id = String( category.id );
+							return (
+								<CheckboxControl
```

Comment:
Ideally this UI would use `<CategoryAutocomplete />` like the category filters in the Inline and Overlay screens. This would improve UI consistency and also avoid massively long category lists.

Reply:
Good idea, changed in a848ab73f2f5a3db55d7e39e7820800b6643061f